src/protocol/utils.rs

```diff
@@ -51,8 +51,13 @@ pub fn binary_search(slots: &Vec<Arc<SlotRange>>, slot: u16) -> Option<Arc<SlotR
   if slot > REDIS_CLUSTER_SLOTS {
     return None;
   }
-
-  let (mut low, mut high) = (0, slots.len() - 1);
```

Comment:
If `slots.len()` is 0, subtracting 1 directly causes a panic in debug builds, but causes an underflow in release builds. The underflow creates an issue later....